File: CCC 2011/WhoHasSeenTheWind.java
import java.util.*;

public class WhoHasSeenTheWind {
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int humidity = Integer.parseInt(in.nextLine());
        int maxTime = Integer.parseInt(in.nextLine());
        boolean landed = false;

        for (int i = 1; i < maxTime; i++){
            if (((-6 * (i * i * i * i) ) + (humidity * (i * i * i)) + (2 * (i * i)) + i) <= 0){
                System.out.println("The ballon first touches ground at hour:\n" +  i);
                landed = true;
                break;
            }
        }
        if (!landed){
            System.out.println("The ballon does not touch ground in the given time.");
        }
    }
}

File: CCC 2011/WhoHasSeenTheWind.cpp
#include <iostream>

using namespace std;

int pow(int num, int pow){
    int current = num;
    for (int i = 0; i < (pow - 1); i++){
        current *= num;
    }
    return current;
}
int main()
{
    int MAX, H;
    bool found = false;
    int at = 0;
    cin >> H >> MAX;
    for (int t = 1; t <= MAX; t++){
        int A = -6 * (pow(t, 4)) + H * pow(t, 3) + 2 * pow(t, 2)+ t;
        if (A <= 0){
            found = true;
            at = t;
            break;
        }
    }
    if (found){
        cout << "The balloon first touches ground at hour:\n" << at << endl;
    } else{
        cout << "The balloon does not touch ground in the given time." << endl;
    }
    return 0;
}